helpers.js

```diff
@@ -0,0 +1,124 @@
+'lang sweet.js';
+
+const TypeCodes = {
+  Identifier: 0,
+  Keyword: 1,
+  Punctuator: 2,
+  NumericLiteral: 3,
+  StringLiteral: 4,
+  TemplateElement: 5,
+  Template: 6,
+  RegExp: 7
+};
+
+function check(obj, type, val) {
+  let isType = obj &&
+    obj.type === 'RawSyntax' &&
+    obj.value.token.typeCode === type;
+
+  if (isType && typeof val !== 'undefined') {
+    if (obj.value.token.typeCode === TypeCodes.StringLiteral) {
+      return obj.value.token.str === val;
+    }
+    return obj.value.token.value === val;
+  }
+  return isType;
+}
+
+export function unwrap(obj) {
+  let hasTok = obj && obj.value && obj.value.token;
+  if (hasTok && obj.value.token.typeCode === TypeCodes.StringLiteral) {
+    return {
+      value: obj.value.token.str
+    };
+  } else if (hasTok && obj.value.token.typeCode !== TypeCodes.Template) {
+    return {
+      value: obj.value.token.value
+    };
+  } else if (hasTok && obj.value.token.typeCode === TypeCodes.Template) {
+    return {
+      value: obj.value.token.items
+    };
+  } else if (obj && obj.type === 'RawDelimiter') {
+    return {
+      value: obj.inner
+    };
+  }
+  return {};
+}
+
+export function isIdentifier(obj, val) {
+  return check(obj, TypeCodes.Identifier, val);
+}
+
+export function fromIdentifier(obj, x) {
+  return obj.value.fromIdentifier(x);
+}
+
```

Comment:
I'm wondering about the order of parameters here. It might be nice to swap them so I can:

```js
const isFor = H.isKeyword.bind('for');
const makeFor = H.fromKeyword.bind('for');
```